OpenAPI/api-contract.yaml

```diff
@@ -0,0 +1,136 @@
+openapi: 3.0.1
+info:
+  title: Maintenance API - Event Driven Internal
+  description: |
+    # Overview
+    Maintenance API - Event Driven Internal is an internal API used in the event driven solution for Maintenance API. 
+
+    The consumer facing documentation is available at https://equinor.github.io/maintenance-api-event-driven-docs/
+  version: "0.1.0"
+  contact:
+    name: Maintenance API Team - TDI FOIT CON PDP
+    url: https://github.com/equinor/maintenance-api-docs
+  
+  license:
+    name: MIT
+    url: http://www.opensource.org/licenses/mit-license.html 
+  
+servers:
+  - url: https://api-dev.gateway.equinor.com/maintenance-api-event-driven-internal
+    description: Test environment  
+  - url: https://api-test.gateway.equinor.com/maintenance-api-event-driven-internal
+    description: QA environment
+  - url: https://api.gateway.equinor.com/maintenance-api-event-driven-internal
+    description: Production environment
+
+paths:
+  /maintenance-events:
+    options:
+      summary: Maintenance events - Handshake
+      description: |
+        ### Overview
+        Handshake 
+      operationId: MaintenanceEventHandshake
+      parameters:   
+      - in: header
+        name: WebHook-Request-Origin
+        description: Hostname for the event publisher - Must match allow list for environment in order to be successful
+        required: true
+        schema:
+          type: string
+      responses:
+        '200':
+          description: Handshake successful
+          headers:
+            'WebHook-Allowed-Origin':
+               description: Allowed origin (should match what value of `WebHook-Request-Origin` in request)
+               schema:
+                type: string
+        '403':
+          description: Handshake unsuccessful
+    post:
+      summary: Maintenance events - Publish
+      description: |
+        ### Overview
+        Publish maintenance event.
+      operationId: MaintenanceEventPublish
+      responses:
+        '200':
+          description: Maintenance event published
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MaintenanceEventPublish'
```

Comment:
im not sure but I feel this should be the request body? and response 201 maybe?

Reply:
Stemmer. Jeg som har blandet her og skal fikse